chrome/content/zotero/xpcom/itemPaneManager.js

```diff
@@ -0,0 +1,340 @@
+/*
+	***** BEGIN LICENSE BLOCK *****
+
+	Copyright © 2024 Corporation for Digital Scholarship
+					 Vienna, Virginia, USA
+					 https://digitalscholar.org
+
+	This file is part of Zotero.
+
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero.  If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
+
+/**
+ * @typedef SectionIcon
+ * @type {object}
+ * @property {string} icon - Icon URI
+ * @property {string} [darkIcon] - Icon URI in dark mode. If not set, use `icon`
+ * @typedef SectionL10n
+ * @type {object}
+ * @property {string} l10nID - data-l10n-id for localization
+ * @property {string} [l10nArgs] - data-l10n-args for localization
+ * @typedef SectionButton
+ * @type {object}
+ * @property {string} type - Button type, must be valid DOMString and without ","
+ * @property {(props: SectionEventHookArgs) => void} onClick - Button click callback
+ * @typedef SectionBasicHookArgs
+ * @type {object}
+ * @property {string} paneID - Registered pane id
+ * @property {Document} doc - Document of section
+ * @property {HTMLDivElement} body - Section body
+ * @typedef SectionUIHookArgs
+ * @type {object}
+ * @property {Zotero.Item} item - Current item
+ * @property {string} tabType - Current tab type
+ * @property {boolean} editable - Whether the section is in edit mode
+ * @property {(l10nArgs: string) => void} setL10nArgs - Set l10n args for section header
+ * @property {(l10nArgs: string) => void} setEnabled - Set pane enabled state
+ * @property {(summary: string) => void} setSectionSummary - Set pane section summary
+ * @property {(buttonType: string, options: {disabled?: boolean; hidden?: boolean}) => void} setSectionButtonStatus - Set pane section button status
```

Comment:
What is `buttonType`?

Reply:
It's the same as 
https://github.com/windingwind/zotero/blob/dad2adeb87ead19ffac363f263f1a2ee41c7e445/chrome/content/zotero/xpcom/itemPaneManager.js#L38

Clicking this button will dispatch an Event of the same name (e.g. `add` is what we use in our sections).

So if you define several different buttons, you can set the status by its `buttonType`.

Reply:
How do you define buttons?

Reply:
```js
sectionButtons: [
		{
			type: "test",
			icon: "chrome://zotero/skin/16/universal/book.svg",
			onClick: ({ item, paneID }) => {
				Zotero.debug("Section clicked!", item?.id);
				Zotero.ItemPaneManager.unregisterSection(paneID);
			},
		},
	],
```